modules/tinymce/src/themes/silver/main/ts/ContextToolbar.ts

```diff
@@ -52,11 +53,26 @@ const anchorOverrides = {
 };
 
 // On desktop we prioritise north-then-south because it's cleaner, but on mobile we prioritise south to try to avoid overlapping with native context toolbars
-const desktopAnchorSpecLayouts = {
-  onLtr: () => [ Layout.north, Layout.south, Layout.northeast, Layout.southeast, Layout.northwest, Layout.southwest,
-    LayoutInside.north, LayoutInside.south, LayoutInside.northeast, LayoutInside.southeast, LayoutInside.northwest, LayoutInside.southwest ],
-  onRtl: () => [ Layout.north, Layout.south, Layout.northwest, Layout.southwest, Layout.northeast, Layout.southeast,
-    LayoutInside.north, LayoutInside.south, LayoutInside.northwest, LayoutInside.southwest, LayoutInside.northeast, LayoutInside.southeast ]
+
+const getDesktopAnchorSpecLayouts = (contextbar: AlloyComponent) => {
```

Comment:
So that it doesn't just jump back to the top as soon as it can, the pinned layouts "remember" their current state and prefer it. They just do that by putting the preference list in the order so that the first preferred pinned state is the current state. Pretty hackily identified.

Reply:
Also, the others haven't been changed yet (mobile etc.)